Sources/MultiSets/CountedSet/CountedSet.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// An unordered, counted multiset.
+@frozen
+public struct CountedSet<Element: Hashable>: RawRepresentable {
```

Comment:
I think `RawRepresentable` makes it very clear how the collection works, and how it can be handled. It also provides a wide variety of useful protocol conformances. I don’t see any reason not to use it, at least for an implicit data structure.

If you’re adamant about removing that conformance, I’d still disagree with `elements` as a name: it includes the elements _and_ their multiplicities.

Reply:
Just because a counted multiset uses a dictionary as its primary storage, it doesn't necessary follows that it needs to be `RawRepresentable` with the dictionary as its raw value.

I find the raw representable terminology misleading in this case -- it implies some kind of interoperability use case which simply doesn't exist here. I don't feel strongly about the property name, apart from objecting to `rawValue`.  Your point against `elements` feels valid & fair!

We definitely do not want to inherit any behavior from RawRepresentable here. (It provides pragmatic but somewhat questionable hashing/== implementations that are geared towards C/ObjC interoperability. We do not want to apply those on this type, even if they happen to work -- rather, we should explicitly define them.)

Reply:
The reason I lean towards `RawRepresentable` is that this entire data structure effectively _is_ a dictionary, rather than simply being backed by one. The formal definition of a multiset is a mapping between a set and nonnegative integers. This is the same relationship an enumeration case has with its raw value: they are wholly interchangeable, but one has added contextual significance.

Anyway, how about `multiplicities` as the property name?